infra/dockerfile.base

```diff
@@ -1,17 +1,21 @@
+FROM python:3.10.7-slim AS builder
+
+COPY . /tmp/bugbug
+
 FROM python:3.10.7-slim
 
 # Setup dependencies in a cacheable step
 COPY requirements.txt /requirements.txt
 RUN apt-get update && \
-		apt-get install -y --no-install-recommends gcc g++ libgomp1 libffi-dev libjemalloc2 zstd patch && \
-		pip install --disable-pip-version-check --quiet --no-cache-dir -r /requirements.txt && \
-		apt-get purge -y gcc g++ libffi-dev patch && \
-		apt-get autoremove -y && \
-		rm -rf /var/lib/apt/lists/*
+	apt-get install -y --no-install-recommends gcc g++ libgomp1 libffi-dev libjemalloc2 zstd patch && \
+	pip install --disable-pip-version-check --quiet --no-cache-dir -r /requirements.txt && \
+	apt-get purge -y gcc g++ libffi-dev patch && \
+	apt-get autoremove -y && \
+	rm -rf /var/lib/apt/lists/*
 
 ENV LD_PRELOAD="libjemalloc.so.2"
 
 COPY infra/mozci_config.toml /root/.config/mozci/config.toml
 
-COPY . /tmp/bugbug
-RUN pip install --disable-pip-version-check --quiet --no-cache-dir /tmp/bugbug
+RUN --mount=target=/tmp/bugbug,from=builder,source=/tmp/bugbug,rw \
+	pip install --disable-pip-version-check --quiet --no-cache-dir /tmp/bugbug
```

Comment:
Nice solution! We could use it directly without having a multi-stage build or doing any `COPY`. See: https://docs.docker.com/build/guide/mounts/#add-bind-mounts